### ISSUE TITLE
`filename-case` false positive for files inside `__mocks__` directory

### ISSUE THREAD

**Issue by DesselBane** [sindresorhus/eslint-plugin-unicorn#2642]
In `jest` and `vitest` it is possible to automatically mock modules by placing files inside a `__mocks__` directory. These files must be called exactly the same as the module they are mocking. Currently the `filename-case` rule via the `configs.all` config reports on these files. 

I would argue this is always incorrect as the user has no control over the filename.
As the ignores property of the rule only considers the filename (AFAIK) and not the entire path the rule needs to be deactivated via the eslint config like so 

```typescript
return [
    eslintPluginUnicorn.configs.all,
    {
      files: ['**/__mocks__/**/*'],
      rules: {
        'unicorn/filename-case': 'off',
      },
    },
  ]
```

This should be the default for `configs.all` / `configs.recommended` IMO

**Comment by DesselBane** [comment]
Hm good point. I guess then everyone does have to configure it
